src/net/polkadot.ts

```diff
@@ -184,4 +211,57 @@ export default class Polkadot extends Api {
       throw new Error('Poldotjs API endpoint not Connected');
     }
   }
+
+  public static async lockDot(
+    polkadotApi: ApiPromise,
+    ethAddress: string,
+    polkadotAddress: string,
+    amount: string,
+    callback: (result: any) => void,
+  ): Promise<any> {
+    const account = await this.getAccount(polkadotAddress);
+
+    if (account) {
+      // to be able to retrieve the signer interface from this account
+      // we can use web3FromSource which will return an InjectedExtension type
+      const injector = await web3FromSource(account.meta.source);
+
+      return polkadotApi?.tx.dot.lock(
+        BASIC_CHANNEL_ID,
+        ethAddress,
+        amount,
+      )
+        .signAndSend(account.address, { signer: injector.signer }, callback);
+    }
+
+    throw new Error('Failed locking dot');
+  }
+
+  public static unlockDot(
+    appDotContract: Contract,
+    ethAddress: string,
+    polkadotAddress: string,
+    amount: string,
+  ): PromiEvent<Contract> {
+    try {
+      const amountWrapped = Web3.utils.toBN(amount);
+      const recipientBytes = ss58ToU8(polkadotAddress!);
+
+      return appDotContract?.methods.burn(
+        recipientBytes,
+        amountWrapped,
+        //  TODO: use incentivized channel ID?
```

Comment:
yeh - lets keep basic for now and wait on this until incentivised channel is completely done, as I suppose it will change the UI/UX somewhat too as it will require the user to have currency to pay fees.